### ISSUE TITLE
comment references: false positive on Markdown links

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/linter#574]
http://dart-lang.github.io/linter/lints/comment_references.html

The `[rfc]` bits reference the inline-URL, but come up as

`INFO: Only reference in scope identifiers in doc comments. ([crypto] lib/src/sha256.dart:15)`

```dart
/// An instance of [Sha256].
///
/// This instance provides convenient access to the [Sha256][rfc] hash function.
///
/// [rfc]: http://tools.ietf.org/html/rfc6234
final sha256 = new Sha256._();
```

**Comment by srawlins** [comment]
Similarly, shortcut reference links are allowed in Markdown as well:

```dart
/// [foo] is a thing. See [here].
///
/// [here]: http://url
final foo;
```

**Comment by natebosch** [comment]
> Similarly, shortcut reference links are allowed in Markdown as well:

I think this should be:

```dart
/// [foo] is a thing. See [here][].
///
/// [here]: http://url
final foo;
```
